uaclient/jobs/eol_status.py

```diff
@@ -0,0 +1,54 @@
+import os
+from typing import Dict
+
+from uaclient import messages, util
+from uaclient.config import UAConfig
+
+
+def get_os_release_vars() -> Dict:
+    vars = dict()
+    content = util.load_file("/etc/os-release")
+    for line in content.splitlines():
+        k, v = line.rstrip().split("=")
+        vars[k] = v.strip('"')
+    return vars
+
+
+def check_eol_and_update(cfg: UAConfig) -> bool:
+    if cfg.is_attached:
+        False
+    os_vars = get_os_release_vars()
+    version_id = os_vars.get("VERSION_ID", "NO-VERSION_ID")
+    codename = os_vars.get(
+        "UBUNTU_CODENAME", "NO-UBUNTU_CODENAME-" + version_id
+    )
+    to_eol, err = util.subp(
+        ["ubuntu-distro-info", "--series", codename, "-yeol"]
+    )
+    days_to_eol = int(to_eol)
+    if days_to_eol >= 1:
+        return True
```

Comment:
Nice I see this is a pretty direct port of the postinst. A lot of this is already done in `util` though. So this can be replaced with something like
```suggestion
    series = util.get_platform_info()["series"]
    if not util.is_active_esm(series)
        return True
```

Reply:
Thanks for this.